### PR TITLE
Delete transaction with tags

### PR DESCRIPTION
**Pull request by Merkur39**
Remove "transaction" relationships on all tags before destroy the transaction.

### CODE REVIEW

File: src/components/useDocuments.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+import { TAGS_DOCTYPE } from 'src/doctypes'
+
+const useDocuments = (doctype, ids) => {
+  const client = useClient()
+  const docs = useSelector(state =>
+    ids.map(id => getDocumentFromState(state, doctype, id))
+  )
+  const hydrated = useMemo(
+    () => client.hydrateDocuments(TAGS_DOCTYPE, docs),
```
Comment (PolariTOON) [inline src/components/useDocuments.js:15]: You should use the `doctype` parameter instead there

File: src/ducks/transactions/TransactionModal/DeleteTransactionRow.jsx [hunk 2]
```diff
@@ -11,12 +11,22 @@ import TrashIcon from 'cozy-ui/transpiled/react/Icons/Trash'
...
+import { TAGS_DOCTYPE } from 'doctypes'
+
 const DeleteTransactionRow = ({ transaction }) => {
   const { t } = useI18n()
   const client = useClient()
   const [showingDeleteConfirmation, setShowDeleteConfirmation] = useState(false)
   const [deleting, setDeleting] = useState(false)
 
+  const transactionTagsIds = getTransactionTags(transaction).map(t => t._id)
+  const transactionTagsWthTransactions = useDocuments(
```
Comment (PolariTOON) [inline src/ducks/transactions/TransactionModal/DeleteTransactionRow.jsx:25]: W**i**th?

**Review by PolariTOON — APPROVED** [review]

**Comment by PolariTOON** [comment]
Perhaps put the deletion logic behind the flag?

File: src/doctypes.js [hunk 3]
```diff
@@ -110,51 +110,51 @@ export class HasManyReimbursements extends HasManyInPlace {
...
 export class HasManyTags extends HasMany {
-  add(docsArg) {
+  async add(docsArg) {
     const docs = Array.isArray(docsArg) ? docsArg : [docsArg]
     const targets = Array.isArray(this.target) ? this.target : [this.target]
 
     for (const tag of docs) {
-      tag.transactions?.add(targets)
+      await tag.transactions?.add(targets)
     }
```
Comment (Crash--) [inline src/doctypes.js:119]: nit for later myself: one way to optimize is to await promiseAll() here.

File: src/doctypes.js [hunk 4]
```diff
@@ -383,3 +383,13 @@ export const buildTagsQueryByIds = ids => {
...
   }
 }
+
+export const buildTransactionsQueryByIds = ids => {
```
Comment (Crash--) [inline src/doctypes.js:387]: buildTransactionsWithTagsQueryByIds

File: src/ducks/transactions/helpers.js [hunk 5]
```diff
@@ -337,3 +337,6 @@ export const hasTag = (transaction, tag) =>
   getTransactionTags(transaction)?.some(
     transactionTag => transactionTag._id === tag._id
   )
+
+export const removeTransaction = (tag, transaction) =>
```
Comment (Crash--) [inline src/ducks/transactions/helpers.js:341]: no async ?

File: src/ducks/transactions/TransactionModal/DeleteTransactionRow.jsx [hunk 6]
```diff
@@ -28,7 +38,11 @@ const DeleteTransactionRow = ({ transaction }) => {
   const handleConfirmDeleteTransaction = async () => {
     try {
       setDeleting(true)
-      await client.destroy(transaction)
+      const { data: lastTransactionRev } = await removeTag(
+        transaction,
+        transactionTagsWthTransactions
+      )
+      await client.destroy(lastTransactionRev)
```
Comment (Crash--) [inline src/ducks/transactions/TransactionModal/DeleteTransactionRow.jsx:45]: nit: I don't like this API so much for now ^^'

**Review by Crash-- — APPROVED** [review]

**Pull request merged**